dataSourceDI/src/main/java/com/panhuk/datasourcedi/di/SessionToken/SessionTokenComponent.kt

```diff
@@ -1,22 +1,28 @@
-package com.panhuk.datasourcedi.di
+package com.panhuk.datasourcedi.di.SessionToken
 
 import com.panhuk.api.di.sessionTokenApi.SessionTokenApiComponent
+import com.panhuk.datasource.DatastorePreferences
 import com.panhuk.datasource.SessionTokenCache
-import com.panhuk.datasource.SessionTokenReader
+import com.panhuk.datasource.SessionTokenDSReader
+import com.panhuk.datasourcedi.di.Api
+import com.panhuk.datasourcedi.di.Cache
+import com.panhuk.datasourcedi.di.DaggerSessionTokenComponent
+import com.panhuk.datasourcedi.di.Datastore.DatastoreComponent
 import dagger.Component
 
 @Component(
   dependencies = [
-    SessionTokenApiComponent::class
+    SessionTokenApiComponent::class,
+    DatastoreComponent::class
   ],
   modules = [
     SessionTokenModule::class
   ]
 )
 interface SessionTokenComponent {
 
-  @Api fun sessionTokenApiReader(): SessionTokenReader
-  @Cache fun sessionTokenCacheReader(): SessionTokenReader
+  @Api fun sessionTokenApiReader(): SessionTokenDSReader
+  @Cache fun sessionTokenCacheReader(): SessionTokenDSReader
   fun sessionTokenCache(): SessionTokenCache
 
   @Component.Builder
```

Comment:
Component builder should contain setter for the DataStoreComponent. Create Pull requests only for code which builds successfully